framework/include/utils/MooseUtils.h

```diff
@@ -809,6 +810,22 @@ struct IsLikeReal<DualReal>
   static constexpr bool value = true;
 };
 
+/**
+ * Custom type trait that has a ::value of true for types that can be broadcasted
+ */
+template <typename T>
+struct canBroadcast
+{
+  static constexpr bool value = std::is_base_of<TIMPI::DataType, TIMPI::StandardType<T>>::value ||
+                                std::is_same<T, std::string>::value;
+};
+template <typename T>
+struct canBroadcast<std::vector<T>>
+{
+  static constexpr bool value = std::is_base_of<TIMPI::DataType, TIMPI::StandardType<T>>::value ||
+                                std::is_same<T, std::string>::value;
+};
```

Comment:
```suggestion
/**
 * Custom type trait that has a ::value of true for types that can be broadcasted
 */
template <typename T>
struct canBroadcast
{
  static constexpr bool value = std::is_base_of<TIMPI::DataType, TIMPI::StandardType<T>>::value ||
                                std::is_same<T, std::string>::value;
};
template <typename T>
struct canBroadcast<std::vector<T>> : canBroadcast<T>
{
};
```

Reply:
Can't do this because it will result in a recursive call and we can't broadcast vector of vectors. At least from my understanding.